### DISCUSSION TITLE
[FEATURE]: delete dependency files in ilogtail/vendor

### DISCUSSION THREAD

**Discussion by bigdata-ny** [alibaba/ilogtail#43] [Ideas]
in ilogtail/vendor,  some dependency files occupy large spaces, eg. k8s.io, golang.org.etc.

![image](https://user-images.githubusercontent.com/20499269/145572910-dd971921-23ac-401f-96de-3dc83617cc17.png)

i suggest delete theses. it's not important in ilogtail.

**Comment by EvanLjp** [comment]
1. ilogtail use vendor to compile rather than pure go modules because [some reasons](https://github.com/alibaba/ilogtail/tree/main/external).
2. Quick start is so complex for our users when delete the packages, may be have the following steps. Also many users would waiting for downloads the packages. 
```
1. git clone codes
2. make vendor to generate vendor
3. go build -mod=vendor
```
For myself, I don't want to delete this packages because iLogtail is not a pure go module project. @shabicheng How do you think about it ?

**Comment by bigdata-ny** [comment]
It seems to make sense. but it keep code clear.
if you want to be simple for users, maybe you can use docker or binary package.

**Comment by EvanLjp** [comment]
Since the current compilation mode is vendor mode, there is no way to remove it for the time being. The issue would be still open until finding an elegant way to solve it.
